pkg/smokescreen/smokescreen.go

```diff
@@ -440,61 +433,69 @@ func getRole(config *Config, req *http.Request) (string, error) {
 	case IsMissingRoleError(err) && config.AllowMissingRole:
 		return "", nil
 	default:
+		config.Log.WithFields(logrus.Fields{
+			"error": err,
+			"is_missing_role": IsMissingRoleError(err),
+			"allow_missing_role": config.AllowMissingRole,
+		}).Error("Unable to get role for request")
 		return "", err
 	}
 }
 
-func checkIfRequestShouldBeProxied(config *Config, req *http.Request, outboundHost string) (*aclDecision, error) {
+func checkIfRequestShouldBeProxied(config *Config, req *http.Request, outboundHost string) (*aclDecision) {
 	decision := &aclDecision{}
 
 	if config.EgressAcl == nil {
 		decision.allow = true
-		decision.reason = "no egress ACL configured"
-		return decision, nil
+		decision.reason = "Egress ACL is not configured"
+		return decision
 	}
 
 	role, roleErr := getRole(config, req)
 	if roleErr != nil {
-		return nil, roleErr
+		decision.reason = "Client role cannot be determined"
+		return decision
 	}
 
 	decision.role = role
 	decision.project, _ = config.EgressAcl.Project(role)
 
 	submatch := hostExtractRE.FindStringSubmatch(outboundHost)
+	destination := submatch[1]
 
-	result, err := config.EgressAcl.Decide(role, submatch[1])
+	action, err := config.EgressAcl.Decide(role, destination)
 	if err != nil {
-		if rerr, ok := err.(UnknownRoleError); ok {
-			//XXX TODO confirm this is still right
-			var msg string
-			if roleErr != nil {
-				msg = fmt.Sprintf("unable to extract a role from your request and no default is provided (%s)", err.Error())
-			} else {
-				msg = fmt.Sprintf("you passed an unknown role '%s'", rerr.Role)
-			}
-			decision.reason = msg
-			return decision, nil
+		config.Log.WithFields(logrus.Fields{
+			"error": err,
+			"role": role,
+		}).Warn("EgressAcl.Decide returned an error.")
+		if role != "" {
+			decision.reason = "Role is invalid or unknown"
+		} else {
+			decision.reason = "Default rules are not set"
 		}
-		return nil, err
+		return decision
 	}
 
-	switch result {
+	switch action {
 	case EgressAclDecisionDeny:
-		decision.reason = "role is not allowed to access this host"
-		return decision, nil
+		decision.reason = "Role is not allowed to access this host"
 
 	case EgressAclDecisionAllowAndReport:
-		decision.reason = "role is not allowed to access this host but report_only is true"
+		decision.reason = "Role is not allowed to access this host but report_only is true"
```

Comment:
Shall we fmt.Sprintf the said role?

Reply:
It would make it easier to look entries up in short term log

Reply:
It could also be another PR. (I can get on that)

Reply:
I had some vague FUD about echoing the role, like if a downstream service lets a user control it, and then displays the proxy's response in a web page or something injectable.  Admittedly, smokescreen probably isn't the weakest link there.

Reply:
If you want to do it, I'll approve it.